components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -229,10 +229,16 @@ protected String getUserInfoEndpoint(OAuthClientResponse token, Map<String, Stri
 
             for (Map.Entry<String, Object> data : jsonObject.entrySet()) {
                 String key = data.getKey();
-                Object value = data.getValue();
+                Object valueObject = data.getValue();
 
-                if (value != null) {
-                    claims.put(ClaimMapping.build(key, key, null, false), value.toString());
+                if (valueObject != null) {
+                    String value;
```

Comment:
simply add the claim inside the if/else and avoid unncessary **_name_** variable